packages/app/src/app/components/CreateNewSandbox/NewSandboxModal/Explore/Explore.tsx

```diff
@@ -35,20 +36,23 @@ export const Explore = () => {
     if (query || category) {
       setPage(0);
     }
-    index
-      .search({
-        facetFilters: ['custom_template.published: true', category],
-        hitsPerPage: 50,
-        query,
-        page,
-      })
-      .then(rsp => {
-        const newTemplates = makeTemplates(rsp.hits);
-        if (page === 0) return setTemplates(newTemplates);
+    if (page <= allPages) {
+      index
+        .search({
+          facetFilters: ['custom_template.published: true', category],
+          hitsPerPage: 50,
+          query,
+          page,
+        })
+        .then(rsp => {
+          setAllPages(rsp.nbPages);
+          const newTemplates = makeTemplates(rsp.hits);
+          if (page === 0) return setTemplates(newTemplates);
```

Comment:
Put curly braces around body, to make this less  error-prone

```suggestion
          if (page === 0) {
            return setTemplates(newTemplates);
          }
```